pkg/asset/installconfig/azure/validation_test.go

```diff
@@ -430,6 +440,11 @@ func TestAzureInstallConfigValidation(t *testing.T) {
 	}
 	azureClient.EXPECT().GetVirtualMachineSku(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, nil).AnyTimes()
 
+	for key, value := range vmCapabilities {
+		azureClient.EXPECT().GetVMCapabilities(gomock.Any(), key, validRegion).Return(value, nil).AnyTimes()
+	}
+	azureClient.EXPECT().GetVMCapabilities(gomock.Any(), "Dne_D2_v4", validRegion).Return(nil, nil).AnyTimes()
```

Comment:
Inside `GetVMCapabilities` you return an error if the `instanceType` is `nil`. So, when mocking, you need to do the same thing here because the test is expecting to match the error message.
```suggestion
	azureClient.EXPECT().GetVMCapabilities(gomock.Any(), "Dne_D2_v4", validRegion).Return(nil, fmt.Errorf("not found in region %s", validRegion).AnyTimes()
```

Reply:
Another point regarding this is, since your `GetVMCapabilities` is now used as `GetVirtualMachineSku`, you need to make sure that all combinations of `instanceType` and `region` expected by the tests are properly mocked.